src/loaders/lottie/tvgLottieProperty.h

```diff
@@ -507,16 +511,88 @@ struct LottiePathSet : LottieProperty
     }
 
 
-    bool operator()(float frameNo, Array<PathCommand>& cmds, Array<Point>& pts, Matrix* transform, LottieExpressions* exps)
+    bool operator()(float frameNo, Array<PathCommand>& cmds, Array<Point>& pts, Matrix* transform, float roundness, LottieExpressions* exps)
     {
         if (exps && (exp && exp->enabled)) {
             if (exp->loop.mode != LottieExpression::LoopMode::None) frameNo = _loop(frames, frameNo, exp);
-            if (exps->result<LottiePathSet>(frameNo, cmds, pts, transform, exp)) return true;
+            if (exps->result<LottiePathSet>(frameNo, cmds, pts, transform, roundness, exp)) return true;
         }
-        return operator()(frameNo, cmds, pts, transform);
+        return operator()(frameNo, cmds, pts, transform, roundness);
     }
 
     void prepare() {}
+
+private:
+    void _roundCorner(Array<PathCommand>& cmds, Array<Point>& pts, const Point& prev, const Point& curr, const Point& next, float roundness)
+    {
+        auto lenPrev = mathLength(prev - curr);
+        auto rPrev = lenPrev > 0.0f ? 0.5f * mathMin(lenPrev * 0.5f, roundness) / lenPrev : 0.0f;
+        auto lenNext = mathLength(next - curr);
+        auto rNext = lenNext > 0.0f ? 0.5f * mathMin(lenNext * 0.5f, roundness) / lenNext : 0.0f;
+
+        auto dPrev = rPrev * (curr - prev);
+        auto dNext = rNext * (curr - next);
+
+        pts.push(curr - 2.0f * dPrev);
+        pts.push(curr - dPrev);
+        pts.push(curr - dNext);
+        pts.push(curr - 2.0f * dNext);
+        cmds.push(PathCommand::LineTo);
+        cmds.push(PathCommand::CubicTo);
+    }
+
+    void _handleCorners(Array<PathCommand>& cmds, Array<Point>& pts, Matrix* transform, float roundness)
```

Comment:
We might need to compare the binary size where these functions(handleCorner, roundCorner) are inside the cpp file vs header file.
`size` will help you the diff.

Reply:
moved as statics outside the class